src/Compilers/Extension/SetGlobalProjectCollectionForCPS.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.ProjectSystem;
+using Microsoft.VisualStudio.ProjectSystem.Build;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Roslyn.Compilers.Extension
+{
+    [ExportBuildGlobalPropertiesProvider]
+    [AppliesTo("(" + ProjectCapabilities.CSharp + " | " + ProjectCapabilities.VB + ")" + " & " + ProjectCapabilities.LanguageService)]
+    public class SetGlobalGlobalPropertiesForCPS : StaticGlobalPropertiesProviderBase
+    {
+        [ImportingConstructor]
+        public SetGlobalGlobalPropertiesForCPS(IProjectService projectService, ILocalRegistry2 registry, IProjectThreadingService threadingService)
+            : base(projectService.Services)
+        {
+            Registry = registry;
+            ThreadingService = threadingService;
+        }
+
+        public ILocalRegistry2 Registry { get; }
+        public IProjectThreadingService ThreadingService { get; }
+
+        public override async Task<IImmutableDictionary<string, string>> GetGlobalPropertiesAsync(CancellationToken cancellationToken)
+        {
+            await ThreadingService.SwitchToUIThread();
+
+            Registry.GetLocalRegistryRoot(out var localRegistryRoot);
+            var registryParts = localRegistryRoot.Split('\\');
+
+            if (registryParts.Length >= 4)
+            {
+                var skuName = registryParts[2];
+                var hiveName = registryParts[3];
+                var roslynHive = string.Format(@"{0}.{1}", registryParts[2], registryParts[3]);
```

Comment:
Is there a reason for `string.Format` over interpolation here or is just preference? Either way is fine but if there is a reason I'm just curious what it is.